afs-ws/afs-ws-client-utils/src/main/java/com/powsybl/afs/ws/client/utils/ClientUtils.java

```diff
@@ -39,7 +47,18 @@ public static Client createClient() {
                 .build();
     }
 
-    private static AfsStorageException createServerErrorException(Response response) {
+    private static RuntimeException createServerErrorException(Response response) {
+        try {
+            String javaException = response.getHeaderString("java-exception");
+            if (javaException != null) {
+                Class exceptionClass = Class.forName(javaException);
+                if (REGISTERED_EXCEPTIONS_FORWARDS.contains(exceptionClass)) {
+                    return (RuntimeException) exceptionClass.newInstance();
+                }
+            }
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            LOGGER.warn("Failed to handle registered exception", e);
+        }
         return new AfsStorageException(response.readEntity(String.class));
```

Comment:
I'm not sure this is the best way to fix this issue: should exception be part of the API, and then be shared between client and server? All exception sent to the client, should be available in the client.

I think we should not depend on the REGISTERED_EXCEPTIONS_FORWARDS.

Reply:
The thing is that I don't want to forward every exception occurring on the server (even if the client has the exception class in its classpath).
That's why I made the declaration of forwardable exception explicit and extendable outside of the API. This is then up to the newly added exception to be part of the server and client. The latter being missing would just endup not forwarding the specific exception and falling back to what was before (AfsStorageException)

Reply:
I’m not convinced by your explanation. Let’s see what @sylvlecl or @jonenst think about your proposal

Reply:
To make sure the client and server has the same exceptions in the classpath, we should drop the plugin system since that prevent being consistent between client and server. Yet it would be a shame that we restrict the exceptions being forwarded to the one defined in the api.

Reply:
@mathbagu I've discussed that with @geofjamg and now I've removed the REGISTERED_EXCEPTIONS_FORWARDS so now all runtime exceptions will be forwarded